src/api/questions.js

```diff
@@ -136,6 +136,32 @@ router.get(
   }
 )
 
+// Delete all questions for a particular queue
+router.delete(
+  '/',
+  [requireQueue, failIfErrors],
+  safeAsync(async (req, res, _next) => {
+    const { id: queueId } = res.locals.queue
+    const questions = await Question.findAll({
+      where: {
+        queueId,
+        dequeueTime: null,
+      },
+      order: [['id', 'ASC']],
+    })
+
+    await Promise.all(
+      questions.map(async question => {
```

Comment:
This method is still going to be super inefficient. We should let the database take care of this - that's what it's good at! We should be using something vaguely like this (untested):

```js
Question.update({
  dequeueTime: new Date(),
}, {
  where: {
    queueId,
    dequeueTime: null,
  }
});
```

Reply:
Ohhh ok that makes sense ~ just pushed the changes!

Reply:
Yay, thanks!